components/button/button-mixin.js

```diff
@@ -121,6 +121,18 @@ export const ButtonMixin = superclass => class extends FocusMixin(superclass) {
 		this.removeEventListener('click', this._handleClick, true);
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		const button = this.shadowRoot.querySelector(this.constructor.focusElementSelector);
+		if (!button) return;
+		button.addEventListener('focus', () => {
+			if (this.shadowRoot.querySelector(':focus-visible')) {
```

Comment:
I don't think this is going to work for browsers who don't support `:focus-visible`, even though the CSS has them fall back to `:focus` --  so they could be in a weird state. I don't think it's worth adding more code here to handle that given that support is so good, but did want to call it out.